components/tidb_query_vec_executors/src/slow_hash_aggr_executor.rs

```diff
@@ -288,8 +289,22 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for SlowHashAggregationImp
                         )?;
                         self.group_key_offsets.push(self.group_key_buffer.len());
                     }
-                    // we have checked that group by cannot be a scalar
-                    _ => unreachable!(),
+                    RpnStackNode::Scalar { value, field_type } => {
+                        if self.cached_encoded_result[i].is_none() {
```

Comment:
nit: use `if let Some()` or `match`

Reply:
@sticnarf 
I changed `if None = self.cached_encoded_result[i] { ... }` to this due to the blame from clippy.

May I change to
```rust
match self.cached_encoded_result[i] {
    None => { ... }
    Some{..} => {}
}
```
?

Reply:
`match` is ok. `if let None` should be just `if ...is_none()`. But in most cases we want `if let Some()` to avoid the extra `unwrap`.